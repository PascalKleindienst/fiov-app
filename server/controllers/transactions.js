// ========================================
// TRANSACTIONS CONTROLLER=================
// ========================================

// List all transactions of the current user.
module.exports.list_all_transactions = (req, res) => {
    req.app.locals.models.Transactions.findAll({ //findAndCountAll
        where: { user_id: req.user.id },
        include: ['category']
    })
        .then( (transactions) => {
            res.json(transactions);
        })
        .catch((err) => {
            res.send(err);
        });
};

// Create a new transaction for the current user.
module.exports.create_transaction = (req, res) => {
    const { description = '', note = '', amount = 0, createdAt = null } = req.body;

    req.app.locals.models.Transactions
        .create({ description, note, amount, createdAt, user_id: req.user.id })
        .then((transaction) => {
            res.json(transaction.get());
        })
        .catch((err) => {
            res.send(err);
        });
};
