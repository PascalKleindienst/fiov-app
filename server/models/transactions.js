// ========================================
// Transactions Model =====================
// ========================================
module.exports = (sequelize, DataTypes) => {
    const Transactions = sequelize.define('Transactions', {
        description: DataTypes.STRING,
        note: DataTypes.TEXT,
        amount: DataTypes.INTEGER
    }, {});

    Transactions.associate = function(models) {
        Transactions.belongsTo(models.Categories, {
            foreignKey: 'category_id',
            onDelete: 'CASCADE',
            as: 'category'
        });

        Transactions.belongsTo(models.User, {
            foreignKey: 'user_id',
            onDelete: 'CASCADE',
            as: 'user'
        });
    };
    
    return Transactions;
};