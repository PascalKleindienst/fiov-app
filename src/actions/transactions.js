// ==============================================
// TRANSACTIONS ACTIONS =========================
// ==============================================
import ajax from 'fetchival';

// =============== Add Transaction ===============

// =============== Remove Transaction ===============

// =============== Update Transaction ===============

// =============== Set Transactions ===============
export const SET_TRANSACTIONS = 'SET_TRANSACTIONS';

export const setTransactions = () => {
    return (dispatch) => {
        return ajax('/api/transactions', { mode: 'cors', credentials: 'same-origin' }).get().then((transactions) => {
            dispatch({
                type: SET_TRANSACTIONS,
                transactions
            });
        });
    };
};