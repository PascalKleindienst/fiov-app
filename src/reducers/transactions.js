// ==============================================
// TRANSACTIONS REDUCERS ========================
// ==============================================
import { SET_TRANSACTIONS } from '../actions/transactions';

const transactionsReducerDefaultState = [];

export default (state = transactionsReducerDefaultState, action) => {
    switch (action.type) {
    case SET_TRANSACTIONS:
        return action.transactions;
    default:
        return state;
    }
};
