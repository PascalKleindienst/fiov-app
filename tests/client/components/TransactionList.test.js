// ==============================================
// TRANSACTIONLIST PAGE TEST ====================
// ==============================================
import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import TransactionListContainer, { TransactionList } from '../../../src/components/TransactionList';
import transactions from '../fixtures/transactions';

describe('my transaction list component', () => {
    const mockStore = configureMockStore([thunk]);

    test('should contain transactions', () => {
        const store = mockStore({
            locale: { languages: ['en'] },
            transactions: transactions
        });
        const wrapper = shallow(<TransactionListContainer store={store} />);
        expect(wrapper.props().transactions).toEqual(transactions);
    });

    test('should render correctly for no transactions', () => {
        const wrapper = shallow(<TransactionList transactions={[]} translate={ (msg) => msg } />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render correctly for transactions', () => {
        const wrapper = shallow(<TransactionList transactions={transactions} translate={ (msg) => msg } />);
        expect(wrapper).toMatchSnapshot();
    });
});
