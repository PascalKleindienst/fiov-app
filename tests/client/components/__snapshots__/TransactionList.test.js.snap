// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my transaction list component should render correctly for no transactions 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-mobile"
    >
      transactions.transactions
    </div>
    <div
      className="show-desktop"
    >
      transactions.transaction
    </div>
    <div
      className="show-desktop"
    >
      transactions.amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <span>
        transactions.no_transactions
      </span>
    </div>
  </div>
</div>
`;

exports[`my transaction list component should render correctly for transactions 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-mobile"
    >
      transactions.transactions
    </div>
    <div
      className="show-desktop"
    >
      transactions.transaction
    </div>
    <div
      className="show-desktop"
    >
      transactions.amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <TransactionListItem
      amount={-199}
      createdAt="1970 01 01"
      description="Gum"
      id={1}
      key="1"
      note=""
    />
    <TransactionListItem
      amount={-109500}
      createdAt="1969 12 28"
      description="Rent"
      id={2}
      key="2"
      note=""
    />
    <TransactionListItem
      amount={4500}
      createdAt="1970 01 05"
      description="Credit Card"
      id={3}
      key="3"
      note=""
    />
  </div>
</div>
`;
